pkg/api/v2/reporter/reporter.go

```diff
@@ -192,34 +193,122 @@ func (e ResourceReports) FilterByKind(kind string) ResourceReports {
 	return reports
 }
 
-// ignores warnings
+// refMapAndSortedKeys returns a map of resource refs to resources and a sorted list of resource refs
+// This is used to iterate over the resources in a consistent order.
+// The reports are keyed by references to the resources, so are not sortable.
+// There is no unique key for a resource, so we use the string representation of the name/namespace as the key,
+// and collect all the resources with the same key together.
+// The previous implementation did not guarantee a consistent order when iterating over the resources,
+// so any order used here will be acceptable for backwards compatibility.
+func (e ResourceReports) refMapAndSortedKeys() (map[string][]resources.InputResource, []string) {
+	// refKeys contains all the unique keys for the resources
+	var refKeys []string
+	// refMaps is a map of resource keys to a slice of resources with that key
+	var refMap = make(map[string][]resources.InputResource)
+
+	// Loop over the resources
+	for res := range e {
```

Comment:
nit: I suspect that you considered this approach, but we can remove the second loop if we do something like:

```go
		// Get a string representation of the resource ref. This is not guaranteed to be unique.
		resKey := res.GetMetadata().Ref().String()

		if _, exists := refMap[resKey]; !exists {
		    refKeys = append(refKeys, resKey)
		}

		// Add the resource to the map of resources with the same key
		refMap[resKey] = append(refMap[resKey], res)
```
That being said, I don't know a lot about the performance of slices in golang so if the current approach (i.e. creating refMap after we know the number of refKeys) is meaningfully faster then we should leave it

Reply:
After reviewing the rest of the PR, this is my only comment. Given that this code is used in the control-plane (i.e. is not performance critical) and I expect the length of refKeys to essentially never be larger than ~1000 (in the case of the largest customer environments), I am approving the PR.

Reply:
Lol, had this approach, then backed it out this morning based on David's comment: https://github.com/solo-io/solo-kit/pull/550/files/412405aac0b1d3498c4ee350e8992360f3045dba#r1496090964